guidance/_grammar.py

```diff
@@ -919,14 +869,23 @@ def str_to_grammar(value: str):
     return partial_grammar
 
 
-def _is_string_literal(node: GrammarFunction):
+def _is_string_literal(node: ComposableGrammar) -> bool:
```

Comment:
This would essentially be the [`TypeGuard`](https://peps.python.org/pep-0647/#:~:text=When%20TypeGuard%20is%20used%20to%20annotate) for the proposed `String = Join[Bytes]` class